<launch>
    <arg name="machine" default="monpal"/> 
    <arg name="is_right" default="true"/>
    <arg name="threshold" default="100"/>
    <arg if="$(arg is_right)" name="position" value="right"/>
    <arg unless="$(arg is_right)" name="position" value="left"/> 
    <arg name="pattern_saving_file" default=""/> 
    <arg name="use_history" default="false" />

   <include file="$(find monpal_launches)/launch/machine.launch" />
   <!--node pkg="tf" type="static_transform_publisher" name="world_to_marker" 
     args="0 0 0 0 1.57  world ar_marker 10" /-->


	<node name="arplus_pose_node_$(arg position)" pkg="arplus_pose" type="arplus_node" respawn="false" output="screen" machine="$(arg machine)">
		<param name="multiPatternFile" type="string" value="$(find arplus_pose)/data/markerboard_0.cfg"/>
		<!--param name="multiPatternFile" type="string" value="$(find arplus_pose)/data/markerboard_1.cfg"/-->
		<!--param name="multiPatternFile" type="string" value="$(find arplus_pose)/data/markerboard_0-1-2-4.cfg"/-->
		<param name="marker_width" type="double" value="121.0"/>
		<param name="threshold" type="int" value="$(arg threshold)"/>
		<param name="is_right" type="bool" value="$(arg is_right)"/>


		<remap from="image" to="/monpal_stereo/$(arg position)/image_mono"/>
		<remap from="camera_info" to="/monpal_stereo/$(arg position)/camera_info"/>				
		<remap from="visualization_marker" to="/visualization_marker_$(arg position)"/>
		<remap from="pose" to="/marker_pose_$(arg position)"/>
        
	</node>
	

</launch>
